bin/common.bash

```diff
@@ -228,3 +228,25 @@ check_openstack_credentials() {
         exit 1
     fi
 }
+
+# Compares the expected and actual git state of the kubespray submodule.
+# If they don't match the user will be asked if they want to continue anyway.
+kubespray_version_check(){
+   pushd "${here}/../" || exit
+
+   result=$(git diff kubespray/)
+
+   popd || exit
+
+   if [[ $result ]]; then
+        echo -e log_info "The status of the kubespray git submodule differs from the expected status, either it is on another commit or there are file changes. This can cause unexpected versions to be installed or cause other errors. We recommend that you stop and check what has changed."
+        echo -n -e log_info "Do you want to abort? (y/n): " 1>&2
```

Comment:
This still shouldn't use echo, but only the common functions. 

```suggestion
        log_info "The status of the kubespray git submodule differs from the expected status, either it is on another commit or there are file changes. This can cause unexpected versions to be installed or cause other errors. We recommend that you stop and check what has changed."
        log_info_no_newline "Do you want to abort? (y/n): "
```